std/js/_std/Reflect.hx

```diff
@@ -19,16 +19,30 @@
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
  * DEALINGS IN THE SOFTWARE.
  */
+
+import js.Syntax;
+import js.Lib;
+import haxe.iterators.StringIterator;
+import haxe.iterators.StringKeyValueIterator;
+
 @:coreApi class Reflect {
 
 	@:pure
 	public inline static function hasField( o : Dynamic, field : String ) : Bool {
 		return js.Object.prototype.hasOwnProperty.call(o, field);
 	}
 
-	@:pure
 	public static function field( o : Dynamic, field : String ) : Dynamic {
-		try return o[cast field] catch( e : Dynamic ) return null;
+		var result = try o[cast field] catch( e : Dynamic ) null;
```

Comment:
Not a fan of this change. We don't support this for `Array` too, so why we treat strings differently here?

Reply:
> Should `String.iterator` be available through `Reflect.field(s, "iterator")` ?

> Probably yeah. Array.iterator is
@simn

Reply:
I thought it was... why does it work through Dynamic not but through `Reflect.field`?

Reply:
I think it only works if you unify with `Iterable` or something like that.